components/pages/AccountPage/ElementLine.tsx

```diff
@@ -0,0 +1,76 @@
+import DeleteOutline from "@mui/icons-material/DeleteOutline";
+import { IconButton } from "@mui/material";
+import FormHelperText from "@mui/material/FormHelperText";
+import MenuItem from "@mui/material/MenuItem";
+import Select from "@mui/material/Select";
+import TableCell from "@mui/material/TableCell";
+import TableRow from "@mui/material/TableRow";
+import TextField from "@mui/material/TextField";
+import { SubmitHandler } from "react-hook-form";
+import { useAccountPage } from "~/components/pages/AccountPage/context";
+import { EditAccountElementSchemaType, useEditAccountElementForm } from "~/forms/editAccountElement";
+import { AccountElement, ElementType } from "~/interfaces/accountElement";
+
+interface ElementLineProps {
+  accountElement: AccountElement;
+}
+
+export default function ElementLine({ accountElement }: ElementLineProps) {
+  const { saveEditAccountElement, removeAccountElement } = useAccountPage();
+
+  const {
+    register,
+    formState: { errors },
+    reset,
+    handleSubmit,
+    getValues,
+  } = useEditAccountElementForm({ name: accountElement.name, type: accountElement.type });
+
+  const onSubmit: SubmitHandler<EditAccountElementSchemaType> = async () => {
+    const values = getValues();
```

Comment:
change this approach to values function props if possible. if not please respond